frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -142,7 +142,7 @@ class DashboardInner extends Component {
   };
 
   onSharingClick = () => {
-    this.props.setSharing(true);
+    this.props.setSharing(this.props.isSharing ? false : true);
```

Comment:
I think a more common way to implement "toggle" behavior is to do `setState(!currentState)`, like:

```js
this.props.setSharing(!this.props.isSharing)
```